rs/index/src/collection/collection.rs

```diff
@@ -356,6 +369,16 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
         Ok(())
     }
 
+    pub fn invalidate(&self, user_id: u128, doc_id: u128) -> Result<bool> {
+        let mut res = self.mutable_segment.read().invalidate(user_id, doc_id)?;
+        // Checks if the `flushing` lock is currently held.
+        if self.flushing.try_lock().is_err() {
+            // The lock could not be acquired, so we are flushing
+            res |= self.tmp_mutable_segment.read().invalidate(user_id, doc_id)?;
```

Comment:
what happens if `flusing` is held, but `tmp_mutable_segment` is not set to `new_writable_segment`?

Reply:
Ok maybe `self.tmp_mutable_segment` being an `Option` may help here? Instead of checking flushing flag, just need to check if `self.tmp_mutable_segment.is_some()`, and invalidate if that's the case. But this means we might miss an invalidation. Maybe looping until `tmp_mutable_segment` becomes non null?

Maybe it's better to keep the invalidated ids map here in collection instead of in mutable segment after all. Wdyt?